iot-hub/Samples/device/ConventionBasedOperations/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -72,6 +88,11 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
             _logger.LogDebug($"Subscribe to commands.");
             await _deviceClient.SubscribeToCommandsAsync(HandleCommandsAsync, cancellationToken);
 
+            // For each component, check if the corresponding properties (both writable and reported) are empty on the initial startup.
+            // If so, report the default values with ACK to the hub.
+            await CheckEmptyProperties(Thermostat1, cancellationToken);

```

Comment:
Call `CheckEmptyProperties()` after setting all handlers; otherwise (call this method before `SetConnectionStatusChangesHandler`), reporting default values by calling `CheckEmptyProperties()` will change the connection status into "Connected", so all existing writable properties will not be obtained by `GetWritablePropertiesAndHandleChangesAsync()` and reported to the hub further.